### DISCUSSION TITLE
Possibility to delete assets in asset upload modal

### DISCUSSION THREAD

**Discussion by arifje** [craftcms/cms#12446] [Ideas] [answered]
So we have like 300.000+ assets, mostly images, related to entries, uploaded by our team when creating articles. After a few months it became impossible to manage assets on the assets page due to long loading times. The browser stops working most of the times when we try to access the "images" folder:

![Screenshot 2022-12-13 at 13 39 41](https://user-images.githubusercontent.com/1286410/207324034-8a26bc8d-5dc4-4b9d-ab06-cac902b2938c.png)

Folder structure we use: **"images/{entry_id}/article-images.jpg"**

This results in the fact that we **never** can delete images from the system. This could easily be solved if we could do a delete action in the assets modal popup, as this lists the assets perfectly when working on an entry.

![Screenshot 2022-12-13 at 13 34 28](https://user-images.githubusercontent.com/1286410/207325024-d28c7446-4120-4f2d-ab47-7b26f0221fef.png)

My suggestion would be to create a delete option (or simple action icon) in the modal, so we can remove assets if wrongly uploaded. In the current situation we have a lot of garbage assets.

Running version: Craft 4 Pro

**Comment by brandonkelly** [comment, marked as answer]
I don’t think it makes sense to add the ability to perform destructive actions like deleting assets from the asset selection modal. But I’ve got some good news! 

The performance issues plaguing your Assets index page are going to go away in Craft 4.4, thanks to a bit of a design overhaul that moves subfolders into the element listing, and paginates them alongside your assets. (See the [Rethinking Volume Subfolders](https://craftcms.com/blog/rethinking-volume-subfolders) blog post and PR #12558 for more info.)

**Reply by arifje** [reply]
Good news!

**Reply by brandonkelly** [reply]
3.8 and 4.4 are out with those changes 🎉
